app/src/editor/shared/components/modules/Map.jsx

```diff
@@ -0,0 +1,190 @@
+                    .values(this.positionHistory)
+                tracePoints.forEach((points) => {
+                    remove(points, (p: TracePoint) => pointList.includes(p.id))
+                })
+            }
+
+            if (msg.markerList && msg.markerList.length > 0) {
+                const { markerList } = msg
+
+                markerList.forEach((id) => {
+                    delete this.queuedMarkers[id]
+                    // NOTE: We should not mutate state but for performance reasons
+                    // it is better to do it here directly instead of a setState call.
+                    // This needs to be done before flushMarkerData performs state
+                    // merging.
+                    delete this.state.markers[id]
+                })
+
+                // Update marker data
+                this.flushMarkerData()
+            }
+        } else {
+            console.error('Unknown message type on MapModule:', msg)
+        }
+    }
+
+    getMarkerFromMessage = (msg: Message): Marker => (
+        {
+            id: msg.id,
+            lat: msg.lat,
+            long: msg.lng,
+            rotation: msg.dir,
+            previousPositions: [],
+        }
+    )
+
+    addTracePoint = (id: string, lat: number, long: number, tracePointId: string) => {
+        let posArray = this.positionHistory[id]
+
+        if (posArray == null) {
+            posArray = []
+            this.positionHistory[id] = posArray
+        }
+
+        const tracePoint: TracePoint = {
+            id: tracePointId,
+            lat,
+            long,
+        }
+        posArray.unshift(tracePoint)
+    }
+
+    getModuleOption = (key: string) => (
+        this.props.module.options[key].value
+    )
+
+    getModuleParam = (key: string) => (
+        this.props.module.params
+            .filter((p) => p.name === key)
+            .map((p) => p.value)
+    )
+
+    flushMarkerData = throttle(() => {
+        if (this.unmounted) {
+            return
+        }
+
+        const { queuedMarkers } = this
+        this.queuedMarkers = {}
+
+        this.setState((state) => ({
+            markers: merge(state.markers, queuedMarkers),
+        }))
+    }, UPDATE_INTERVAL_MS)
+
+    render() {
+        const { className } = this.props
+        const { markers } = this.state
+
+        const directionalMarkers = this.getModuleOption('directionalMarkers')
+        let markerIcon = this.getModuleOption('markerIcon')
+        if (directionalMarkers) {
+            markerIcon = this.getModuleOption('directionalMarkerIcon')
+        }
+
+        return (
+            <div className={cx(className)}>
+                <ModuleSubscription
+                    {...this.props}
+                    onMessage={this.onMessage}
+                />
+                <Map
+                    {...this.props.module.options}
+                    className={styles.map}
+                    centerLat={this.getModuleOption('centerLat') || 60.18}
```

Comment:
Can none of these be `0`?

Suggest perhaps changing `this.getModuleOption` to take the default as a second param, and using the default if the key you're asking for is falsey but not `0`?

i.e. 
```js
this.getModuleOption('centerLat', 60.18)
```

```js
getModuleOption = (key: string, defaultValue: any) => {
    const { value } = this.props.module.options[key]
    if (!value && value !== 0) {
        return defaultValue // use default if value falsey but not zero
    }
    return value
}
```